File: 523-continuous-subarray-sum/continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
      int n = nums.length ; 
      int pre =0 ;
      int sum = 0 ;
      int mod = 0 ;
      Set<Integer> set = new HashSet<>();
      for(int i = 0 ; i<n ; i++){
        sum+=nums[i];
        mod = k==0? sum:sum%k;
        if(set.contains(mod)) return true ;
        set.add(pre);
        pre = mod ;
        }
      return false ;
        
    }
}

File: 523-continuous-subarray-sum/continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        int prefixSum = 0 ; 
        int pre = 0 ; 
        set<int> checkMod ; 
        for(int i = 0; i < nums.size() ; i++){
            prefixSum+=nums[i] ; 
            int mod = prefixSum % k ; 
            if(checkMod.find(mod)!=checkMod.end()){
                return true ;
            }
            checkMod.insert(pre);
            pre = mod ;
        }
        return false ;
    }
};